osu.Game/Online/Rooms/OnlinePlayBeatmapAvailabilityTracker.cs

```diff
@@ -58,39 +64,50 @@ protected override void LoadComplete()
 
                 downloadTracker?.RemoveAndDisposeImmediately();
 
-                Debug.Assert(item.NewValue.Beatmap.Value.BeatmapSet != null);
+                beatmapLookupCache.GetBeatmapAsync(item.NewValue.Beatmap.OnlineID).ContinueWith(task => Schedule(() =>
```

Comment:
This has no `cancellationToken`, I think it can lead to a race condition if the `SelectedItem` changes quickly twice or more?

Reply:
The continuation handles this so I didn't think it necessary to add the complexity of a cancellation token.